web/src/compose_validate.ts

```diff
@@ -67,18 +86,31 @@ function set_message_too_long_for_edit(status: boolean, $container: JQuery): voi
 
 export function set_recipient_disallowed(status: boolean): void {
     recipient_disallowed = status;
-    update_send_button_status();
 }
 
 function update_send_button_status(): void {
     $(".message-send-controls").toggleClass(
         "disabled-message-send-controls",
-        message_too_long || upload_in_progress || recipient_disallowed,
+        upload_in_progress ||
+            no_stream_mentioned ||
+            (no_topic_mentioned && compose_state.get_message_type() === "stream") ||
+            (no_private_recipient && compose_state.get_message_type() === "private") ||
+            message_too_long ||
+            recipient_disallowed ||
+            no_message_content,
     );
 }
 
 export function get_disabled_send_tooltip(): string {
-    if (message_too_long) {
+    if (no_stream_mentioned && compose_state.get_message_type() === "stream") {
+        return $t({defaultMessage: "Please select a channel."});
```

Comment:
Can we define these strings as constants so that they are easier to modify in the future?

Reply:
Note that the same string is used in banner text too.

Reply:
See validate_stream_message

Reply:
Yeah I'd expect `validate_stream_message` to be able to call this function to get the strings.